src/service/grpcnode.ts

```diff
@@ -4,12 +4,12 @@ import {FileDescriptorProto} from "google-protobuf/google/protobuf/descriptor_pb
 import {CodeGeneratorResponse} from "google-protobuf/google/protobuf/compiler/plugin_pb";
 import {createFile, RPCDescriptor, GrpcServiceDescriptor, RPCMethodDescriptor} from "./common";
 
-export function generateGrpcNodeService(filename: string, descriptor: FileDescriptorProto, exportMap: ExportMap): CodeGeneratorResponse.File {
+export function generateGrpcNodeService(filename: string, descriptor: FileDescriptorProto, exportMap: ExportMap, modeParameter: string | undefined): CodeGeneratorResponse.File {
```

Comment:
This function is only called from the `index.ts` file so the argument could either be an enum to make the mode explicit.